scripts/utils/printing_tools.js

```diff
@@ -31,6 +31,18 @@ const toErc20Units = function (amount, decimals) {
   return representation
 }
 
+/**
+ * A generalized version of "toWei" for tokens with an arbitrary amount of decimals.
+ * If the decimal representation has more decimals than the maximum amount possible, then the extra decimals are truncated.
+ *
+ * @param {number} amount Float representation for the amount of some ERC20 token
+ * @param {(number|string|BN)} decimals Maximum number of decimals of the token
+ * @returns {BN} number of token units corresponding to the input amount
+ */
+const floatToErc20Units = function (amount, decimals) {
+  return toErc20Units(amount.toString(), decimals)
```

Comment:
I think this function will fail for `amount=10^21` because as a string it's `1e+21`, which is not supported by `toErc20Units`. `amount=10^20` would still work, however. 
It would be good to always print the float with decimal notation, I've no idea how to do that in JS however.

Reply:
Oh, and `toErc20Units` fails if the number of decimals is more than what you would expect from the token. So, if you try to do `toErc20Units("1.111", 2)` it will fail. What I would do here in e.g. Python would be to tell my parser to print exactly `decimals` digits.

Reply:
>  I think this function will fail for amount=10^21

I do not expect amounts to be of this size, but will be happy to put an assertion.

> Oh, and toErc20Units fails if the number of decimals is more than what you would expect from the token.

We also will not experience these issues in this script since the decimals are fixed at 18. Perhaps I should have left the `toWEI` implementation that I already had to keep this script entirely self contained.

It seems to me that you have suggested I use your awesome helper method and then pointed out a bunch of problems with it? Would you agree that I revert the use of this helper method in favour of the self-contained `toWEI`, put more assertions or include some comments about how the edge cases are unexpected to occur in this particular script?

Reply:
We could do that for now, but I'd definitely open an issue about it so that we'll eventually have a reliable way to go from token units to decimal representation and vice versa without relying on some hack. We could extend `toErc20Units` so that it accepts float. Feel free to assign me to it.